charms/worker/k8s/src/charm.py

```diff
@@ -95,14 +96,17 @@ def _cluster_departing_unit(event: ops.EventBase) -> Union[Literal[False], ops.U
     )
 
 
+class NodeRemovedError(Exception):
+    """Raised to prevent reconciliation of dying node."""
+
+
```

Comment:
Once a unit evaluates that it is marked for removal, it should never reconcile afterwards